### PR TITLE
Updating bubble sort code in feature branch

### PR DESCRIPTION
**Pull request by Mindo-Joseph**
### This pull request achieves the following
- Creation of a bubble_sort function that takes in an array of integers and sorts them by comparing them against each other using the bubble sort algorithm
- Creation of a bubble_sort_by function that sorts an array of strings from smallest to largest and takes in a block

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
 # BubbleSort
 Ruby implementation of bubble sort algorithm
+## Built With
```
Comment (majovanilla) [inline README.md]: Improve your Readme file. It should be descriptive of the project. You are using the correct template but now you need to fill it with information about your project. You can add details about what you did in this project, why you did it, how you did it...Check this [link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/) for more info.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#Method definition
+def bubble_sort(array)
+    # Get the length of the array
+    n = array.length
+    # If the size is equal to 1 return the array as it is already sorted
+    if n == 1
+        array
+    # If the size is 0 return a message saying the array is empty
+    elsif n == 0
```
Comment (majovanilla) [inline bubble_sort.rb]: There is no need to include all these comments. Your code should be descriptive enough. Please delete them as requested on the [pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png)

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                if array[i] > array[i+1]
+                    # swap the values
+                    array[i],array[i+1] = array[i+1],array[i]
+                    # update the sorted variable to true
+                    sorted = true
+                else
+                    # Otherwise if it is less than or equal to the next leave them as they are since they are sorted
+                    array[i] <= array[i+1]
+                        array[i], array[i+1] = array[i], array[i+1]
+
```
Comment (majovanilla) [inline bubble_sort.rb]: There is no need to do this. It's like saying "when this happens, do nothing" and if you don't include it, well it won't do it anyway! :smile:

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    
+                
+                    
+                end
+
+
+            }
+            # Break out of the loop if everything evaluates to true
+            break if !sorted
+        }
```
Comment (majovanilla) [inline bubble_sort.rb]: There is a better way to do this loop (the "ruby way") You can create a while loop that runs until your sorted variable is true.

File: bubble_sort.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+def bubble_sort_by(array)
+    # Get the array size
+    n = array.size
+    # If the size is 1 return the array as it is already sorted
+    if n == 1
+        array
+    # If the size of the array is 0 return a string that tells the user the array is empty   
+    elsif n == 0
+        "Array is empty"
+
```
Comment (majovanilla) [inline bubble_sort.rb]: You can do a ternary operator here: `array.empty? ? "Array is empty : array ` Check more info about the ternary operator [here](https://www.rubyguides.com/2019/10/ruby-ternary-operator/).

File: bubble_sort.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                    # Swap the two words
+                    array[i] , array[i+1] = array[i+1],array[i]
+                    # Update sorted variable to be true
+                    sorted = true
+                
+                end 
+                    
+            }
+            # Break out of the loop if sorted is true
+            break if !sorted
```
Comment (majovanilla) [inline bubble_sort.rb]: Same here with the while loop

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hi @Mindo-Joseph @KossySteve 

## Good job on this project! 👍

- Good Pull Request Message
- Nice Readme template

There are still some changes 👇 you need to implement to approve this request 🔍

### Required Changes: ✍️

- Check your linters! You have errors there :smiley: 
- Check the comments on your code below.


## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Review by mosesogwo — APPROVED** [review]
**Hi @KossySteve, @Mindo-Joseph :clap:. Well done with the changes :heavy_check_mark:**

But before you :rocket:, fix these: 
- Use hyperlink in your README for the live link to keep it neat.
- Name of each Author should precede the contact links.
- The correct naming convention should be bubble_sort, not bubblesort. 

**Cheers :stars:**

### Status: Approved!

**Pull request merged**